insights/combiners/tmpfilesd.py

```diff
@@ -0,0 +1,79 @@
+"""
+Tmpfilesd configuration
+=======================
+
+The tmpfilesd files are normally available to rules as a list of TmpFilesd
+objects. This combiner turns those into one set of data, and provides a
+``find_file()`` method to search for that filename among all the files.
+"""
+
+from insights.core.plugins import combiner
+from insights.parsers.tmpfilesd import TmpFilesd
+
+
+@combiner(TmpFilesd)
+class AllTmpFiles(object):
+    """
+    Combiner for accessing all the tmpfilesd configuration files. Configuration
+    files can be found in three directories: ``/usr/lib/tmpfiles.d``,
+    ``/run/tmpfiles.d``, and ``/etc/tmpfiles.d``. Each directory overrides the
+    settings in the previous directory. This combiner checks for and accounts
+    for this behavior.
+
+    Attributes:
+        files(set): the set of files found in all data files.
+        rules(list): the entire list of rules
+        file_paths(list): a list of the file paths for the configurations.
+    """
+    def __init__(self, tmpfiles):
+        rules = []
+        files = set([])
+        file_paths = []
+        self.tmpfiles = tmpfiles
+        rules_found = {}
+        rule_number = 0
+        # Files must be sorted so that /etc/tmpfiles.d overrides /run/tmpfiles.d
+        # which overides /usr/lib/tmpfiles.d. To do this, we have to list file
+        # paths in reverse order.
+        for tmpfiles in sorted(tmpfiles, key=lambda f: f.file_path, reverse=True):
+            for file_ in tmpfiles.rules:
```

Comment:
A little tricky too (to save the `rules_found` and `rule_number`):
Just FYI:
~~~
             for new_rule in tmpfiles.rules:                                                                                                                                                                      
                 _file = (new_rule['path'])                                                                                                                                                                       
                 if _file in files:                                                                                                                                                                               
                      [rules.remove(_rule) for _rule in rules if _rule['path'] == _file]                                                                                                                           
                 rules.append(new_rule)
~~~